infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LogReplicationContext.java

```diff
@@ -97,4 +97,14 @@ public ISerializer getProtobufSerializer() {
         return configManager.getRuntime().getSerializers().getSerializer(PROTOBUF_SERIALIZER_CODE);
     }
 
+    /**
+     * Get the CorfuStore from LogReplicationConfigManager, whose table registry is guaranteed to have log replication
+     * related system tables opened. It is also useful to help perform forced snapshot syncs driven by the sessions
+     * themselves (for example, in the case of group destination change in LOGICAL_GROUP replication).
```

Comment:
yes the java doc is not correct, it should be "client tables, registry table and protobuf descriptor table".

For the metadataManager & configManager runtime consolidation, actually it has been there in master as well, I remember we discussed it a little bit before. I agree we could revisit and see if they could be optimized. It's a bit beyond this change so maybe we could address in separate PR?

Reply:
right, we had discussed about the runtime before. I was talking about the corfuStore...there is one instantiated in metadataManager and one in configManager using the same runtime. 
Was checking if we could instantiate corfuStore, maybe in sessionManager, and then pass it on to the metadataManager & configManager.

I think this change could be small, but if you like we can take this as a separate task. I am fine either ways.
I agree the runtime change can be more intense..

Reply:
oh I see, for CorfuStore yes we do instantiate in both places. but also in both classes there are more than 1 references for CorfuRuntime, which means we gonna pass both CorfuStore and CorfuRuntime as arguments for both classes..

IMO it might be fine to just keep current pattern, as CorfuStore is very thin layer on top of runtime, and CorfuRuntime was only created once.